File: src/Java/Post.java
package Java;


import java.util.Date;
import java.util.List;

public class Post {
    protected Date date;
    protected User author;
    protected String title ;
    protected String content;
    
    protected List<Comment> comments;

    public Post(){
    }

    public Post(String title, String content, User author){
        this.title = title;
        this.content = content;
        this.author = author;
        this.date = new Date();
    }

    // Metodos Getters

    public Date getDate(){
        return this.date;
    }

    public User getAutor(){
        return this.author;
    }

    public String getTitle(){
        return this.title;
    }

    public String getContent(){
        return this.content;
    }


    // Metodos Setters
    public void setTitle(String title){
        this.title = title;
    }

    public void setContent(String content){
        this.content = content;
    }


    @Override
    public String toString(){

        StringBuilder sb = new StringBuilder();
        sb.append("-----------------------------------\n");
        sb.append("Titulo: " + this.title + "\n");
        sb.append("Contenido: " + this.content + "\n");
        sb.append("Autor: " + this.author.getUsername() + "\n");
        sb.append("Fecha: " + this.date + "\n");
        return sb.toString();
    }

    public void addComment(String text, User author){
        Comment comment = new Comment(0, text, author, this);
        this.comments.add(comment);
    }
}

File: src/C#/networking-app/MyApplication/Post.cs
using System.Collections.Generic;
using static System.DateTime;

namespace MyApplication.Models
{
	public class Post
	{
		protected string title;
		protected string content;
		protected User author;
		protected List<Comment> comments;
		protected DateTime date;
		protected int id;
		protected static int contadorId;
		
		private Post()
		{
			this.id = ++Post.contadorId;
		}
		
		public Post(string title, string content, User author) : this()
		{
			this.date = DateTime.Now;
			this.author = author;
			this.title = title;
			this.content = content;
			this.comments = new List<Comment>();
			this.title = title;
		}
		
		// Metodos Getters y Setters
		public int Id
		{
			get {return id;}
			set {id = value;}
		}
		
		public DateTime Date
		{
			get {return date;}
			set {date = value;}
		}
		
		public User Author
		{
			get {return author;}
			set {author = value;}
		}
		
		public string Title
		{
			get {return title;}
			set {title = value;}
		}
		
		public string Content
		{
			get {return content;}
			set {content = value;}
		}
		
		public List<Comment> Comments
		{
			get {return comments;}
			set {comments = value;}
		}
	}
}
